examples/simple_key_value_store.rs

```diff
@@ -0,0 +1,414 @@
+        stats_value.stats.push((0, in_coming_data));
+        Err(InterfaceError::Abort)
+    }
+    fn handle_get_key(&mut self,
+                      _type_id: u64,
+                      _name: NameType,
+                      _our_authority: Authority,
+                      _from_authority: Authority,
+                      _from_address: NameType) -> Result<MessageAction, InterfaceError> {
+        unimplemented!();
+    }
+}
+
+struct TestNodeGenerator;
+
+impl CreatePersonas<TestNode> for TestNodeGenerator {
+    fn create_personas(&mut self) -> TestNode {
+        TestNode::new()
+    }
+}
+
+fn main() {
+    let args = Docopt::new(USAGE)
+                      .and_then(|dopt| dopt.parse())
+                      .unwrap_or_else(|e| e.exit());
+
+    // You can conveniently access values with `get_{bool,count,str,vec}`
+    // functions. If the key doesn't exist (or if, e.g., you use `get_str` on
+    // a switch), then a sensible default value is returned.
+    let node_type = args.get_str("--type");
+    let mut command = String::new();
+    match node_type {
+        "node" => {
+            let mut test_node = RoutingNode::<TestNode, TestNodeGenerator>::new(TestNodeGenerator);
+            if !args.get_str("end_point").is_empty() {
+                match SocketAddr::from_str(args.get_str("end_point")) {
+                    Ok(addr) => {
+                        println!("initial bootstrapping to {} ", addr);
+                        let _ = test_node.bootstrap(Some(vec![Endpoint::Tcp(addr)]), None);
+                    }
+                    Err(_) => {}
+                };
+            } else {
+                // if no bootstrap endpoint provided, still need to call the bootstrap method to trigger default behaviour
+                let _ = test_node.bootstrap(None, None);
+            }
+            loop {
+                command.clear();
+                println!("Input command (stop)");
+                let _ = io::stdin().read_line(&mut command);
+                let v: Vec<&str> = command.split(' ').collect();
+                match v[0].trim() {
+                    "stop" => break,
+                    _ => println!("Invalid Option")
+                }
+            }
+        },
+        "first" => {
+            let mut test_node = RoutingNode::<TestNode, TestNodeGenerator>::new(TestNodeGenerator);
+            test_node.run_zero_membrane();
+            loop {
+                let mut command = String::new();
+                command.clear();
+                println!("Input command (stop)");
+                let _ = io::stdin().read_line(&mut command);
+                let v: Vec<&str> = command.split(' ').collect();
+                match v[0].trim() {
+                    "stop" => break,
+                    _ => println!("Invalid Option")
+              }
+            }
+        },
+        // default to client behaviour
+        _ => {
```

Comment:
here "client" should be explicitly mentioned.  Otherwise, any typo will create client